lib/app/features/feed/providers/bookmarks_notifier.c.dart

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:collection/collection.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/exceptions/exceptions.dart';
+import 'package:ion/app/features/auth/providers/auth_provider.c.dart';
+import 'package:ion/app/features/feed/data/models/bookmarks/bookmarks.c.dart';
+import 'package:ion/app/features/feed/data/models/bookmarks/bookmarks_set.c.dart';
+import 'package:ion/app/features/feed/data/models/entities/article_data.c.dart';
+import 'package:ion/app/features/nostr/model/action_source.dart';
+import 'package:ion/app/features/nostr/model/replaceable_event_reference.c.dart';
+import 'package:ion/app/features/nostr/providers/nostr_cache.c.dart';
+import 'package:ion/app/features/nostr/providers/nostr_notifier.c.dart';
+import 'package:nostr_dart/nostr_dart.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'bookmarks_notifier.c.g.dart';
+
+@Riverpod(keepAlive: true)
```

Comment:
1. Probably not mandatory, but if we don't have it, there will be a cache/relay read on every bookmark button build, so on every article cell build I think, right? 
2. I intentionally return a Map here to avoid additional relay requests for different types. This way we fetch them all at one with a single request (with multiple filters) and keep them in cache, if we added the parameter we would have N relay requests for N types.

Reply:
done